compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -298,6 +298,19 @@ class TypeComparer(@constructorOnly initctx: Context) extends ConstraintHandling
       }
     }
 
+    /** In capture checking, implements the logic to compare type variables which represent
+     *  capture variables.
+     *
+     *  Note: should only be called in a context where tp1 or tp2 is a type variable representing a capture variable.
+     *
+     *  @return -1 if tp1 or tp2 is not a capture variables, 1 if both tp1 and tp2 are capture variables and tp1 is a subcapture of tp2,
```

Comment:
I'd try to move this out of firstTry. firstTry on named types is teh fast path, everything that takes more time should go to thirdTryNamed.